lib/src/model.dart

```diff
@@ -2163,21 +2135,32 @@ abstract class ModelElement extends Nameable
         newModelElement = new Typedef(e, library);
       }
       if (e is FieldElement) {
+        assert(getter != null || setter != null);
         if (enclosingClass == null) {
-          if (index != null) {
-            assert(getter != null);
-            newModelElement =
-                new EnumField.forConstant(index, e, library, getter);
-          } else {
-            if (e.enclosingElement.isEnum) {
-              newModelElement = new EnumField(e, library, getter, setter);
+          if (e.enclosingElement.isEnum) {
+            int index =
+                e.computeConstantValue()?.getField('index')?.toIntValue();
```

Comment:
Oh, that's so much better.  This logic is what I intended: if I had seen/thought to look for e.isEnumConstant I would have written it exactly like this.  Thanks!